File: c#/functions/Callbacks.cs
using System;

class Callbacks {
  public delegate string Validation(double v);

  static double SourceValue(string prompt, Validation validate)
  {
    double value;
    string validationMessage;

    do {
      System.Console.Write(prompt);
      value =  Convert.ToDouble(Console.ReadLine());
      validationMessage = validate(value);

      if (string.IsNullOrEmpty(validationMessage)) {
        break;
      }

      System.Console.WriteLine(validationMessage);
      System.Console.WriteLine("Try again.");
    } while (true);

    return value;
  }

  static string ValidateTemperatureInCelsius(double value)
  {
    string message = "";

    if (value > 26) {
      message = "Temperature is to high for human health.";
    } else if (value < 22) {
      message = "Temperature is to low for human health.";
    }

    return message;
  }

  public static void Main(string[] args) {
    double temperature;
    temperature = SourceValue("Give the ambient temperature in degrees Celsius: ", ValidateTemperatureInCelsius);
    System.Console.WriteLine("Tempetature has been set to " + temperature + " degree Celsius.");
  }
}

File: python/functions/callbacks.py
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md
# SPDX-License-Identifier: GPL-3.0-or-later

import typing

def sourceValue(prompt: str, validate: callable):
  while True:
    value = float(input(prompt))
    validation_message = validate(value)

    if not validation_message: break

    print(validation_message)
    print("Try again.")

  return value

def validateTemperatureInCelsius(value) -> str:
    message = ""

    if (value > 26):
        message = "Temperature is to high for human health."
    elif (value < 22):
        message = "Temperature is to low for human health."

    return message

temperature = sourceValue("Give the ambient temperature in degrees Celsius: ", validateTemperatureInCelsius)
print("Tempetature has been set to ", temperature, " degree Celsius.\n")
